File: test_project/.vscode/SnipHub/snip/hello.swift
// Swift Hello World
import Foundation

// 简单打印
print("Hello, World!")

// 使用函数
func sayHello(to name: String) -> String {
    return "Hello, \(name)!"
}

print(sayHello(to: "Swift"))

// 使用类
class Greeter {
    private let name: String
    
    init(name: String) {
        self.name = name
    }
    
    func greet() -> String {
        return "Hello, \(name)!"
    }
}

// 创建实例并调用方法
let greeter = Greeter(name: "Swift OOP")
print(greeter.greet())

// 使用结构体
struct Person {
    var name: String
    var age: Int
    
    func describe() -> String {
        return "\(name) is \(age) years old."
    }
}

let person = Person(name: "Swift Developer", age: 30)
print(person.describe())

// 使用枚举
enum Language {
    case swift, objectiveC, cSharp, java
    
    func description() -> String {
        switch self {
        case .swift:
            return "Swift is developed by Apple"
        case .objectiveC:
            return "Objective-C is the predecessor of Swift"
        case .cSharp:
            return "C# is developed by Microsoft"
        case .java:
            return "Java is developed by Sun Microsystems (now Oracle)"
        }
    }
}

let myFavoriteLanguage = Language.swift
print(myFavoriteLanguage.description())

File: test_project/.vscode/SnipHub/snip/hello.cpp
// C++ Hello World
#include <iostream>
#include <string>

// 函数声明
std::string sayHello(const std::string& name);

// 类声明
class Greeter {
private:
    std::string name;
    
public:
    Greeter(const std::string& n) : name(n) {}
    
    std::string greet() const {
        return "Hello, " + name + "!";
    }
};

int main() {
    std::cout << "Hello, World!" << std::endl;
    
    // 使用函数
    std::cout << sayHello("C++") << std::endl;
    
    // 使用类
    Greeter greeter("C++ OOP");
    std::cout << greeter.greet() << std::endl;
    
    return 0;
}

// 函数定义
std::string sayHello(const std::string& name) {
    return "Hello, " + name + "!";
}
